### PR TITLE
Initial release to main for v1.0.0

### PR DESCRIPTION
**Pull request by ed-xmos**
All commits have now been reviewed on the merge to develop so hopefully a rubber stamp job but an opportunity to catch any issues if required.

### CODE REVIEW

File: lib_qadc/src/qadc_pot.xc [hunk 1]
```diff
@@ -0,0 +1,555 @@
...
+            ticks = (uint32_t)ticks << QADC_Q_3_13_SHIFT / max_scale_up;
+            // ticks = ((int64_t)max_scale_up * (int64_t)ticks) >> QADC_Q_3_13_SHIFT;
+            
+            uint16_t max = 0;
+            max_arg = num_points - 1;
+            for(int i = num_points - 1; i >= 0; i--){
+                if(ticks > up[i]){
+                    if(up[i] > max){
+                        max_arg = i - 1;
+                        max = up[i];
```
Comment (danielpieczko) [inline lib_qadc/src/qadc_pot.xc:136]: Why is max_args assigned (i-1) when max is set from index i?

File: examples/pot_reader/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+set(APP_COMPILER_FLAGS_CONTINUOUS_MEM   ${COMMON_COMPILER_FLAGS}
+                                        -DCONTINUOUS=1
+                                        -DUSE_SHARED_MEMORY=1)
+
+
+set(APP_COMPILER_FLAGS_SINGLE       ${COMMON_COMPILER_FLAGS}
+                                    -DCONTINUOUS=0)
+
+
+# Workaround for now until cmake xcommon supports this
```
[the remark this answers is not in the document]
Comment (ed-xmos) [reply examples/pot_reader/CMakeLists.txt:25]: no idea! removed..

File: lib_qadc/lib_build_info.cmake [hunk 3]
```diff
@@ -0,0 +1,6 @@
+set(LIB_NAME lib_qadc)
+set(LIB_VERSION 1.0.0)
+set(LIB_INCLUDES api)
+set(LIB_COMPILER_FLAGS -Os)
+
```
[the remark this answers is not in the document]
Comment (ed-xmos) [reply lib_qadc/lib_build_info.cmake:5]: I was unaware of that.. better to be explicit too.

File: lib_qadc/api/qadc_pot.h [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+
+/**
+ * \addtogroup lib_qadc_pot_reader
+ *
+ * The public API for using the QADC.
+ * @{
+ */
+
+/** 
+ * @brief   Macro for sizing the the state array used by QADC. Please declare a state array
```
[the remark this answers is not in the document]
Comment (ed-xmos) [reply lib_qadc/api/qadc_pot.h:45]: Not a tribute to 80s band https://en.wikipedia.org/wiki/The_The

File: lib_qadc/src/qadc_rheo.xc [hunk 5]
```diff
@@ -0,0 +1,389 @@
...
+                do_adc_convert(p_adc, adc_idx, adc_rheo_state, rheo_timings, adc_mode);
+            break;
+            case tmr_single when timerafter(rheo_timings.time_trigger_overshoot) :> int _:
+                do_adc_handle_overshoot(p_adc, adc_idx, adc_rheo_state, rheo_timings);
+            break;
+        }
+        result = adc_rheo_state.results[adc_idx];
+    }
+
+    // The pot value has only just reached the threshold so allow some time for it to nearly reach the pot value
```
[the remark this answers is not in the document]
Comment (ed-xmos) [reply lib_qadc/src/qadc_rheo.xc:383]: voltage is better

File: examples/pot_reader/src/main.xc [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+#define LUT_SIZE            1024
+#define FILTER_DEPTH        16
+#define HYSTERESIS          1
+
+// Select your port as required. Either 1b or wide ports may be used.
+// on tile[1]: port p_adc[] = {XS1_PORT_1M, XS1_PORT_1O}; // Sets which pins are to be used (channels 0..n) X1D36/38
+on tile[1]: port p_adc[] = {XS1_PORT_4A}; // Sets which pins are to be used (channels 0..n) X1D02/03
+
+
+void control_task(chanend ?c_adc, uint16_t * unsafe result_ptr){
```
[the remark this answers is not in the document]
Comment (ed-xmos) [reply examples/pot_reader/src/main.xc:20]: I'm in two minds about this. Ifdeffing is the old XMOS way of doing it. All the init code is common. Will have a think.

At a minimum I have changed ctrl_task to qadc_pot_continuous_example etc.

**Review by danielpieczko — APPROVED** [review]

**Pull request merged**